text/0000-light-dom.md

```diff
@@ -0,0 +1,293 @@
+
+```js
+import { LightningElement } from "lwc";
+
+// Example of a shadow DOM component
+class ShadowDOMComponent extends LightningElement {}
+
+// Example of a light DOM component
+class LightDOMComponent extends LightningElement {
+  static shadowDOM = false;
+}
+
+// Default value
+console.log(LightningElement.shadowDOM); // true
+```
+
+The `shadowDOM` property is looked up by the LWC engine when the component is instantiated for the first time and is cached for future instantiation. Changing the value of the `shadowDOM` static property after the first instantiation doesn't influence whether components are rendered in the light DOM or in the shadow DOM.
+
+It also means that developers should be careful not to override the `shadowDOM` static property value when inheriting from another component. Switching a component mode from shadow DOM to light DOM (and vice-versa) in the child class would certainly break logic in the base class.
+
+```js
+import { LightningElement } from "lwc";
+
+class Base extends LightningElement {}
+
+class Child extends Base {
+  // ⚠️ Changing the shadowDOM static property value in a component class inheriting from a base
+  // will certainly class some unexpected issue.
+  static shadowDOM = false;
+}
+```
+
+### Component features when using Light DOM
+
+Some of the LWC component capabilities are directly inherited from Shadow DOM, or emulated by the synthetic-shadow. Despite the use of Light DOM, we’d like to keep these features available, even if their behavior is slightly adapted to the Light DOM:
+
+#### Slots
+
+As mentioned before, the component composition model in LWC is provided by slots. Light DOM will provide the same mental model for developers building Light DOM components.
+
+In Light DOM, `<slot>` will denote the place where the slotted component will be attached. The `<slot>` element itself won't be rendered. The slotted content (or the fallback content) will be flattened to the parent element at runtime.
+
+Since the `<slot>` element itself isn't rendered, adding attributes or event listeners to the `<slot>` element in the template will throw a compiler error.
+
+Due to the above differences from regular `<slot>`, LWC compiler will enforce the presence of `<slot lwc:no-shadow>` on Light DOM slots to make it explicit to the user that these are different slots.
+
+##### Timing
+
+In terms of timing, slots in light DOM will behave similarly to the current synthetic shadow slots. For example:
+
+```html
+<!-- x/slottable.html -->
+<template lwc:no-shadow>
+  <slot lwc:no-shadow name="foo"></slot>
+  <slot lwc:no-shadow></slot>
+  <slot lwc:no-shadow name="bar"></slot>
+</template>
+```
+
+```html
+<!-- app.html -->
+<template>
+  <x-slottable>
+    <x-baz></x-baz>
+    <x-foo slot="foo"></x-foo>
+    <x-bar slot="bar"></x-bar>
```

Comment:
Now that I think about this, I don't think the `slot` attribute should be rendered at runtime in the light DOM. 

We decided not to render the `<slot>` element at runtime to prevent potential issues where the light DOM `<slot>`  connects to the parent shadow root. If we were to keep the `slot` attribute at runtime for the light DOM we could also run into unexpected slotting behaviors at runtime when mixing shadow DOM slot and light DOM slots.

Reply:
Does this run into the same problem where the consumer component needs to know whether `<x-slottable>` is a light DOM element or not? If so, it would maybe argue for `no-shadow:slot="foo"` or `lwc:slot="foo"` or something like that.

Then again as you said above, that does not solve default slots. Unless we force something like `lwc:slot="default"`.